deeplearning4j/deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/layers/SubsamplingLayer.java

```diff
@@ -102,7 +102,9 @@ protected SubsamplingLayer(BaseSubsamplingBuilder builder) {
         this.convolutionMode = builder.convolutionMode;
         if (builder instanceof Builder) {
             this.dilation = ((Builder) builder).dilation;
-            this.cnn2dDataFormat = ((Builder) builder).dataFormat;
+            this.cnn2dDataFormat = builder.cnn2DFormat;
+        } else {
+            this.cnn2dDataFormat = builder.cnn2DFormat;
```

Comment:
why an if/else here? you are doing the same thing, put it before this if clause.

Reply:
Ah, updated. Originally there was an if for pooling 2d since it was missing for the subsampling 1d layer. Now I just added it as a general property. That was more just an artifact. Updated.